src/main/java/org/apache/sling/jcr/resource/internal/helper/jcr/JcrResourceProvider.java

```diff
@@ -358,28 +358,21 @@ public Iterator<Resource> listChildren(ResolveContext<JcrProviderState> ctx, Res
     @Override
     public @Nullable Resource getParent(final @NotNull ResolveContext<JcrProviderState> ctx, final @NotNull Resource child) {
         if (child instanceof JcrItemResource<?>) {
-            try {
-                String version = null;
-                if (child.getResourceMetadata().getParameterMap() != null) {
-                    version = child.getResourceMetadata().getParameterMap().get("v");
-                }
-                if (version == null) {
-                    String parentPath = ResourceUtil.getParent(child.getPath());
-                    if (parentPath != null) {
-                        Item parentItem = ctx.getProviderState().getResourceFactory()
-                            .getItemOrNull(parentPath);
-                        if (parentItem != null && parentItem.isNode()) {
-                            return new JcrNodeResource(ctx.getResourceResolver(),
-                                parentPath, null, (Node)parentItem,
-                                ctx.getProviderState().getHelperData());
-                        }
+            String version = null;
+            if (child.getResourceMetadata().getParameterMap() != null) {
+                version = child.getResourceMetadata().getParameterMap().get("v");
+            }
+            if (version == null) {
+                String parentPath = ResourceUtil.getParent(child.getPath());
```

Comment:
If we knew here that it is already a Jackrabbit-Repo, we would not need to calculate the parentPath for this codepath. This would be only necessary when line 371 is reached.

Reply:
not quite.... but i fell into the same trap..... but looking at ResourceUtil.getParent it does some extra magic like returning null if the item is the root node.
also the calculated parent path is used for the fallback when no JackrabbitSession is available.
so, i decided to leave the code instead of having it duplicated with all the logic involved.... but maybe there is a way to simplify it.... but ResourceUtil.getParent looks quite complex and i didn't want to introduce nasty side effects by replacing it by PathUtil.getParent or some other utility i am familiar with.